### ISSUE TITLE
Validation issue: seeResponseContainsJson() validates certain invalid data

### ISSUE THREAD

**Issue by DanTalash** [Codeception/Codeception#2070]
I have an issue with validating some data containing an array of integers using the `seeResponseContainsJson()` function. This is using the latest version of the library (last updated today)

I am making a request that returns the following response (JSON)
````JSON
{
    "responseCode": 0,
    "message": "OK",
    "data": [
        9,
        0,
        0
    ]
}
````

I cannot validate it by using `seeResponseContainsJson()` if the expected data is an array of three zeros, as shown below:

````PHP
//Incorrectly passes validation, even though the first integer does not match.
$I->seeResponseContainsJson([
	'responseCode' => 0,
	'data' => [
		0,
		0,
		0,
	]
]);

//Correctly fails validation if any of the zeros are changed to a non-zero (or string) value
$I->seeResponseContainsJson([
	'responseCode' => 0,
	'data' => [
		1,
		0,
		0,
	]
]);
````

It is pretty straight forward to perform this validation by hand, but it is quite odd that the validation erroneously passes in some cases.

**Comment by Naktibalda** [comment]
I reproduced the issue.
It isn't limited to zeroes, but happens to any value which is repeated multiple times.

It is caused by a way in which Codeception\Util\JsonArray [compares sequential arrays](https://github.com/Codeception/Codeception/blob/37af30da1566a31e00dcdf76e1b0046153dced61/src/Codeception/Util/JsonArray.php#L107)

```php
    public function testContainsArrayMatchesArrayWithValueRepeatedMultipleTimesCorrectly()
    {
        $jsonArray = new JsonArray(json_encode(['foo', 'foo', 'bar']));
        $expectedArray = ['foo', 'foo', 'foo'];
        $this->assertFalse($jsonArray->containsArray($expectedArray));       
    }
```

sequentialArrayIntersect iterates the expected array and checks if each value is present in result,
so [0] matches expected result [0,0,0]

Also it intentionally ignores position of the value, so [1, 2, 3] matches [3, 2, 1]

**Comment by Naktibalda** [comment]
I improved JsonArray to compare repeated values correctly.

**Comment by Naktibalda** [comment]
I merged the patch to 2.0 branch
